### PR TITLE
Add missing unit in metadata.yaml

### PR DESCRIPTION
**Pull request by mackjmr**
**Description:** 
Metric unit is required, see: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/cmd/mdatagen/metadata-schema.yaml#L72-L73.

This PR adds the unit to all metrics where this is missing.

I have inferred the unit from the metric name, and would appreciate that someone more familiar with the receivers in question review the chosen units.

### CODE REVIEW

File: receiver/redisreceiver/metadata.yaml [hunk 1]
```diff
@@ -103,7 +104,7 @@ metrics:
   redis.clients.connected:
     enabled: true
     description: Number of client connections (excluding connections from replicas)
-    unit:
+    unit: "{connections}"
```
Comment (djaglowski) [inline receiver/redisreceiver/metadata.yaml:107]:
```suggestion
    unit: "{clients}"
```
Looks like we're counting clients here (with a filter).

File: receiver/redisreceiver/metadata.yaml [hunk 2]
```diff
@@ -112,21 +113,21 @@ metrics:
...
-    unit: ""
+    unit: "By"
     gauge:
       value_type: int
 
   redis.clients.max_output_buffer:
     enabled: true
     description: Longest output list among current client connections
-    unit: ""
+    unit: "By"
```
Comment (djaglowski) [inline receiver/redisreceiver/metadata.yaml:123]: Technically these two are breaking changes - will you please submit these as a separate PR?

File: receiver/redisreceiver/metadata.yaml [hunk 3]
```diff
@@ -200,14 +201,14 @@ metrics:
   redis.memory.fragmentation_ratio:
     enabled: true
     description: Ratio between used_memory_rss and used_memory
-    unit: ""
+    unit: "{ratio}"
```
Comment (djaglowski) [inline receiver/redisreceiver/metadata.yaml:204]: Not sure I agree with this one. "ratio" describes the value in a way, but it's not a unit of measurement.

File: receiver/redisreceiver/metadata.yaml [hunk 4]
```diff
@@ -250,7 +251,7 @@ metrics:
   redis.keyspace.hits:
     enabled: true
     description: Number of successful lookup of keys in the main dictionary
-    unit: ""
+    unit: "{lookups}"
```
Comment (djaglowski) [inline receiver/redisreceiver/metadata.yaml:254]:
```suggestion
    unit: "{hits}"
```

File: receiver/redisreceiver/metadata.yaml [hunk 5]
```diff
@@ -259,7 +260,7 @@ metrics:
   redis.keyspace.misses:
     enabled: true
     description: Number of failed lookup of keys in the main dictionary
-    unit: ""
+    unit: "{lookups}"
```
Comment (djaglowski) [inline receiver/redisreceiver/metadata.yaml:263]:
```suggestion
    unit: "{misses}"
```

File: receiver/redisreceiver/metadata.yaml [hunk 6]
```diff
@@ -275,7 +276,7 @@ metrics:
   redis.slaves.connected:
     enabled: true
     description: Number of connected replicas
-    unit: ""
+    unit: "{connections}"
```
Comment (djaglowski) [inline receiver/redisreceiver/metadata.yaml:279]:
```suggestion
    unit: "{replicas}"
```

File: receiver/redisreceiver/metadata.yaml [hunk 7]
```diff
@@ -284,29 +285,29 @@ metrics:
...
-    unit: ""
+    unit: "By"
     gauge:
       value_type: int
 
   redis.replication.offset:
     enabled: true
     description: The server's current replication offset
-    unit: ""
+    unit: "By"
```
Comment (djaglowski) [inline receiver/redisreceiver/metadata.yaml:295]: These also are breaking changes. Will you open a dedicated PR for these?

File: receiver/redisreceiver/metadata.yaml (hunk 2 above)
Comment (mackjmr) [reply receiver/redisreceiver/metadata.yaml]: Thanks for the review! Removed from this PR and opened a separate PR with a `changelog` here: #23454.

File: receiver/redisreceiver/metadata.yaml (hunk 7 above)
Comment (mackjmr) [reply receiver/redisreceiver/metadata.yaml]: Also added in other PR: #23454

File: receiver/redisreceiver/metadata.yaml (hunk 3 above)
Comment (mackjmr) [reply receiver/redisreceiver/metadata.yaml:204]: Agreed, ratios are `unitless` -- perhaps I should put "{no unit: ratio}" ?

Comment (djaglowski) [reply receiver/redisreceiver/metadata.yaml]: 👍 I believe "1" is the default unit, but it's ok to just omit it.

Comment (mackjmr) [reply receiver/redisreceiver/metadata.yaml]: I'm working on validation of the `metadata.yaml` ([issue](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/23424)), and one of the checks that I currently have makes sure that the unit is not empty because it is a required field. This is an edge case, as the metric is unitless.  
I've made a commit to put the unit to 1 for `redis.memory.fragmentation_ratio` instead of leaving it empty -- let me know if this is OK ?

Comment (dmitryax) [reply receiver/redisreceiver/metadata.yaml]: I think 1 is the correct unit for ratio

**Comment by mackjmr** [comment]
Thanks for the explanation @djaglowski, I've updated the golden files 👍

**Comment by dmitryax** [comment]
@mackjmr, can you please split the PR into one per component and add changelog entries?

**Comment by mackjmr** [comment]
@dmitryax split into 1 PR per component with changelog:
- redisreceiver: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/23573
- aerospikereceiver: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/23572
- postgresqlreceiver: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/23571

Closing this PR.
